block/produce.go

```diff
@@ -186,10 +186,17 @@ func (m *Manager) produceBlock(allowEmpty bool) (*types.Block, *types.Commit, er
 		return nil, nil, fmt.Errorf("apply block: %w: %w", err, ErrNonRecoverable)
 	}
 
+	// validate configuration params and rollapp consensus params keep in line
+	err = m.ValidateConfigWithRollappParams()
+	if err != nil {
+		panic(err)
```

Comment:
i think it's better to return here the error wrapping ErrUnrecoverable. why?
1. In general we handle errors in the higher level callee 
2. we don't really want to panic cause that will get us in a restart loop which won't help, we want to have the higher level caller catch it and emit unhealthy event.

Reply:
i removed the panic here, but then i added panic if error after applying in the p2p onReceivedBlock, otherwise the error will just be logged and can be missed.

Reply:
yea so I see we call `applyBlock` (in case of full node):
1. from da 
2. from p2p

seems like we're panicing here, and don't go into unhealthy mode - though we probably should - as I don't see the benefit of constant panicing. 

@srene  @danwt wdyt.

Reply:
makes sense 👍

Reply:
i replaced panic by health status event